back-end/src/sciler/communication/communication.go

```diff
@@ -38,23 +45,32 @@ func NewCommunicator(host string, port int, topicsOfInterest []string, handler m
 	opts.SetPingTimeout(timeout)             // time after sending a PING request to the broker
 	opts.SetMaxReconnectInterval(timeout)    // max time before retrying to reconnect
 	opts.SetConnectTimeout(20 * time.Second) // time before timing out and erroring the attempt
+}
 
-	opts.SetOnConnectHandler(func(client mqtt.Client) {
-		// on connect subscribe and execute onStart
-		topics := make(map[string]byte)
-		for _, topic := range topicsOfInterest {
-			topics[topic] = byte(0)
-		}
+// setHandlers is a method that configures what will be done on connect, disconnect and reconnect
+func setHandlers(opts *mqtt.ClientOptions, messageHandler mqtt.MessageHandler, onStart func()) {
+	opts.SetOnConnectHandler(func(client mqtt.Client) { // on connect subscribe and execute onStart
 		action(func() mqtt.Token {
-			return client.SubscribeMultiple(topics, handler)
+			return client.Subscribe("back-end", 2, messageHandler)
```

Comment:
document that we only subscribe to back-end now

Reply:
done